app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/test/chapter1/actions/RecyclerViewActionsTest.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.android.architecture.blueprints.todoapp.test.chapter1.actions
+
+import android.support.test.espresso.Espresso
+import android.support.test.espresso.action.ViewActions
+import android.support.test.espresso.contrib.RecyclerViewActions
+import android.support.test.espresso.matcher.ViewMatchers
+import android.support.v7.widget.RecyclerView
+import com.example.android.architecture.blueprints.todoapp.R
+import com.example.android.architecture.blueprints.todoapp.test.BaseTest
+import com.example.android.architecture.blueprints.todoapp.test.chapter2.customactions.CustomClickAction
+import com.example.android.architecture.blueprints.todoapp.test.chapter2.customactions.CustomRecyclerViewActions
+import com.example.android.architecture.blueprints.todoapp.test.chapter2.customactions.CustomRecyclerViewActions.ClickTodoCheckBoxWithTitleViewAction
+import com.example.android.architecture.blueprints.todoapp.test.chapter4.conditionwatchers.ConditionWatchers.waitForElement
+import com.example.android.architecture.blueprints.todoapp.test.chapter4.conditionwatchers.ConditionWatchers.waitForElementIsGone
+import org.junit.Test
+
+/**
+ * Demonstrates [RecyclerView] actions usage.
+ */
+class RecyclerViewActionsTest : BaseTest() {
+    private val todoSavedSnackbar = Espresso.onView(ViewMatchers.withText(R.string.successfully_saved_task_message))
```

Comment:
It is better and a bit easier to read when you import `onView` instead of using `Espresso.onView` every time. The same with `withText` instead of using `ViewMatchers.withText`. Why? You have a clear separation between built in view matchers / view actions and custom one you may implement. And it usually means less code which improves readability by a lot.